gcp/telemetry.ts

```diff
@@ -0,0 +1,257 @@
+import * as https from "https";
+import * as fs from "fs";
+import * as path from "path";
+import AdmZip from "adm-zip";
+
+import * as config from "./config";
+import * as pulumi from "@pulumi/pulumi";
+import * as k8s from "@pulumi/kubernetes";
+import * as gcp from "@pulumi/gcp";
+
+class Dashboard {
+  name: string;
+  json: string;
+
+  constructor(name: string, json: string) {
+    this.name = name;
+    this.json = json;
+  }
+
+  filename(): string {
+    return `${this.name}.json`;
+  }
+}
+
+export class Backends {
+  provider: k8s.Provider;
+  cluster: gcp.container.Cluster;
+
+  constructor(provider: k8s.Provider, cluster: gcp.container.Cluster) {
+    this.provider = provider;
+    this.cluster = cluster;
+  }
+
+  install(): void {
+    new Grafana().install(this.provider, this.cluster);
+    new Prometheus().install(this.provider, this.cluster);
+  }
+}
+
+class Prometheus {
+  install(k8sProvider: k8s.Provider, cluster: gcp.container.Cluster): void {
+    new k8s.helm.v3.Chart(
+      "prometheus",
+      {
+        chart: "prometheus",
+        fetchOpts: {
+          repo: "https://prometheus-community.github.io/helm-charts",
+        },
+        // Prometheus defaults are good enough for Lightbend Telemetry, so we don't need to customize values here.
+        // If you need to change something, you can check the available chart values by running:
+        // $ helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+        // $ helm show values prometheus-community/prometheus
+      },
+      {
+        provider: k8sProvider,
+        customTimeouts: { create: "30m", update: "30m", delete: "30m" },
+        dependsOn: [cluster],
+      },
+    );
+  }
+}
+
+class Grafana {
+  private static readonly DashboardsFile = `cinnamon-grafana-prometheus-${config.Telemetry.Version}.zip`;
+  private static readonly DashboardsUrl = `https://downloads.lightbend.com/cinnamon/grafana/cinnamon-grafana-prometheus-${config.Telemetry.Version}.zip`;
+  private static readonly DashboardDownloadDir: string = `${path.resolve()}/downloads`;
+
+  private createDashboardDirectory(): Promise<string> {
```

Comment:
there is a lot of common logic here, it would be nice to be shared

Reply:
Let's handle it at https://github.com/lightbend/akka-cloud-platform-deploy/issues/18.